src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,165 @@
+                        && Objects.equals(current.key, key)) {
+                    return current.value;
+                } else {
+                    current = current.next;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        return "table=" + Arrays.toString(table);
+    }
+
+    @SuppressWarnings({"unchecked"})
+    private void resize() {
+        Node<K,V>[] oldTable = table;
+        table = (Node<K,V>[]) new Node[oldTable.length * DOUBLING_FACTOR];
+        size = 0;
+        transfer(oldTable);
+    }
+
+    private void appendNode(Node<K, V> current, Node<K, V> newNode) {
+        current.next = newNode;
+        size++;
+    }
+
+    private boolean areKeysEqual(K currentKey, K newKey) {
+        return hash(currentKey) == hash(newKey)
+                && (Objects.equals(currentKey, newKey));
+    }
+
+    private void changeValue(Node<K, V> current, Node<K, V> newNode) {
+        current.value = newNode.value;
+    }
+
+    private void transfer(Node<K, V>[] oldTable) {
+        for (Node<K, V> oldNode : oldTable) {
+            if (oldNode != null) {
+                Node<K, V> current = oldNode;
+                while (current != null) {
+                    put(current.key, current.value);
+                    current = current.next;
+                }
+            }
+        }
+    }
+
+    private int hash(Object key) {
+        return (key == null) ? 0 : key.hashCode();
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == this) {
+                return true;
+            }
+            if (obj == null) {
+                return false;
+            }
+            if (obj.getClass().equals(Node.class)) {
+                Node<K,V> current = (Node<K,V>) obj;
+                return ((hash == current.hash)
+                        && Objects.equals(key, current.key)
+                        && Objects.equals(value, current.value)
+                        && Objects.equals(next, current.next));
+            }
+            return false;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = 17;
+            result = 31 * result + hash;
+            result = 31 * result + (key == null ? 0 : key.hashCode());
+            result = 31 * result + (value == null ? 0 : value.hashCode());
+            result = 31 * result + (next == null ? 0 : next.hashCode());
+            return result;
+        }
```

Comment:
we don't need equals/hashcode for nodes, we compare keys of nodes directly anyway